src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -504,13 +513,13 @@ private fun RsMod.insertExternCrateItem(psiFactory: RsPsiFactory, crateName: Str
     }
 }
 
-private fun RsMod.insertUseItem(psiFactory: RsPsiFactory, usePath: String) {
+private fun RsItemsOwner.insertUseItem(psiFactory: RsPsiFactory, usePath: String) {
     val useItem = psiFactory.createUseItem(usePath)
     if (tryGroupWithOtherUseItems(psiFactory, useItem)) return
     val anchor = childrenOfType<RsUseItem>().lastElement ?: childrenOfType<RsExternCrateItem>().lastElement
     if (anchor != null) {
         val insertedUseItem = addAfter(useItem, anchor)
-        if (anchor is RsExternCrateItem) {
+        if (anchor is RsExternCrateItem || isDoctestInjection) {
```

Comment:
Add a comment that we have to insert newline manually because autoformatting doesn't work in injected code